src/Limiters/quasimonotone.jl

```diff
@@ -158,7 +222,54 @@ function compute_neighbor_bounds_local!(
     limiter::QuasiMonotoneLimiter,
     ρ,
     ::ClimaComms.CUDADevice,
-) end
+)
+    topology = Spaces.topology(axes(ρ))
+    Ni, Nj, _, Nv, Nh = size(Fields.field_values(ρ))
+    nthreads, nblocks = config_threadblock(Nv, Nh)
+    local_neighbor_elem_offset =
+        Adapt.adapt(CuVector{Int}, topology.local_neighbor_elem_offset)
+    local_neighbor_elem =
+        Adapt.adapt(CuVector{Int}, topology.local_neighbor_elem)
```

Comment:
This will do the conversion from a CPU array to a GPU array at every time step. Would be better to change this in the topology itself (AFAIK this isn't used by anything else)

Reply:
Also, I thought the plan was to change the limiters to be element-local in any case?

Reply:
> This will do the conversion from a CPU array to a GPU array at every time step. Would be better to change this in the topology itself (AFAIK this isn't used by anything else)

We spoke about this last week, and I think you suggested to not adapt topology because it is such a huge struct. Regarding the element-local, I think that was specific to the water borrowing implementation. Being that this occurs after DSS, it's not clear to me why we would need any distributed support anyway. Want me to just delete `compute_neighbor_bounds_local!` and all of the ghost stuff?

Reply:
> I think we spoke about this last week, and I think you suggested to not adapt topology because it is such a huge struct.

You don't want to adapt topology itself, but you can make the `local_neighbor_elem`/`local_neighbor_elem_offset` fields Arrays/CuArrays depending on the device

Reply:
> Regarding the element-local, I think that was specific to the water borrowing implementation. Being that this occurs after DSS, it's not clear to me why we would need any distributed support anyway. Want me to just delete `compute_neighbor_bounds_local!` and all of the ghost stuff?

Maybe that's easiest? we would have to rework all the examples.

Reply:
Ah, ok, and add them to the `QuasiMonotoneLimiter` struct? Sure, will do

Reply:
the only difference is how you compute `min`/`max`, so we could make that an option.

Reply:
> Ah, ok, and add them to the `QuasiMonotoneLimiter` struct? Sure, will do

No, i meant make them parametric inside the topology

Reply:
I'm not sure I follow, make what parametric inside the topology struct?

Reply:
> Maybe that's easiest? we would have to rework all the examples.

Why would we need to rework the examples?

Reply:
@simonbyrne, where are you suggesting `local_neighbor_elem_offset` and `local_neighbor_elem` live?

Reply:
Ah, I see, so just adapt those pieces, sounds good